com.hunterwittenborn.Celeste.yml

```diff
@@ -0,0 +1,54 @@
+app-id: com.hunterwittenborn.Celeste
+runtime: org.gnome.Platform
+runtime-version: "43"
+sdk: org.gnome.Sdk
+command: celeste
+finish-args:
+  # We need to display on Wayland/X11.
+  - '--socket=wayland'
+  - '--socket=fallback-x11'
+  # Users may want to sync files from anywhere on their system.
+  - '--filesystem=host'
+  # We need network access in order to sync files to remotes.
+  - '--share=network'
+  # Used for X11 performance as described at https://docs.flatpak.org/en/latest/sandbox-permissions-reference.html#f1.
+  - '--share=ipc'
+  # DBus connections we use.
+  - '--talk-name=org.kde.StatusNotifierWatcher'
+  - '--talk-name=com.canonical.AppMenu.Registrar'
+  - '--talk-name=com.canonical.indicator.application'
+  # Extra libraries that we've bundled with the Flatpak in order to run.
+  - '--env=LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu:/snap/gnome-3-38-2004/119/usr/lib/x86_64-linux-gnu'
+modules:
+  - name: unsquashfs
+    buildsystem: simple
+    build-commands:
+      - "sed -i 's|#XZ_SUPPORT|XZ_SUPPORT|' squashfs-tools/Makefile"
+      - 'cd squashfs-tools/ && make unsquashfs'
+      - 'install -Dm 755 squashfs-tools/unsquashfs /app/bin/unsquashfs'
+    sources:
+      - type: 'git'
+        url: 'https://github.com/plougher/squashfs-tools'
+        tag: '4.5.1'
+  - name: celeste
+    buildsystem: simple
+    build-commands:
+      # Unpack and install the Snap package and other related files.
+      - 'unsquashfs celeste.snap'
+      - 'cp squashfs-root/* /app/ -R'
+      - 'install -Dm 644 com.hunterwittenborn.Celeste.metainfo.xml /app/share/metainfo/com.hunterwittenborn.Celeste.metainfo.xml'
+
+      # Remove unwanted icon files.
+      - 'cd /app/usr/share/icons/ && find ./ -mindepth 1 -maxdepth 1 -not -name hicolor -exec rm -R {} +'
```

Comment:
if they are unwanted why are they in the snap package?

Reply:
Flatpak seems to do some stuff where it shares the needed icons across apps. It's not of my knowledge that Snaps always do that - if I recall there was an issue with the Snap where icons wouldn't appear on some systems due to helper Snaps not being connected or something, so it was decided the icons would get bundled. These don't work in the Flatpak though, and attempting to include them shows a bunch of warning about them not being exported.

Reply:
flatpak will **export** the icons matching the app-id so that they can be used to display the app, along with the appstream and desktop files. We have things  in place like `rename-icon` that takes care of rename to the right name *if needed*.

Reply:
it's good now.